From c927fb9d220474ef1bee021f6fb9bf22bae42ef1 Mon Sep 17 00:00:00 2001
From: Martin Magr <mmagr@redhat.com>
Date: Tue, 26 Aug 2014 17:54:12 +0200
Subject: [PATCH] Fixed manage_service feature

Using manage_service=>false in class neutron::agents::ovs generates
error. This patch is fixing the issue and makes mentioned use-case
installable.

Change-Id: I5a3205c3e21e1312463f47d938f5854a189c5d38
Closes-bug: #1359823
---
 manifests/agents/ovs.pp | 24 +++++++++++++-----------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/manifests/agents/ovs.pp b/manifests/agents/ovs.pp
index 0b4401a..2db7a1a 100644
--- a/manifests/agents/ovs.pp
+++ b/manifests/agents/ovs.pp
@@ -137,17 +137,19 @@ class neutron::agents::ovs (
     } else {
       $service_ensure = 'stopped'
     }
-    service { 'neutron-plugin-ovs-service':
-      ensure  => $service_ensure,
-      name    => $::neutron::params::ovs_agent_service,
-      enable  => $enabled,
-      require => Class['neutron'],
-    }
-    if $::neutron::params::ovs_cleanup_service {
-      service {'ovs-cleanup-service':
-        name   => $::neutron::params::ovs_cleanup_service,
-        enable => $enabled,
-      }
+  }
+
+  service { 'neutron-plugin-ovs-service':
+    ensure  => $service_ensure,
+    name    => $::neutron::params::ovs_agent_service,
+    enable  => $enabled,
+    require => Class['neutron'],
+  }
+
+  if $::neutron::params::ovs_cleanup_service {
+    service {'ovs-cleanup-service':
+      name   => $::neutron::params::ovs_cleanup_service,
+      enable => $enabled,
     }
   }
 
-- 
1.9.3

